src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -56,23 +56,23 @@ internal sealed class TextBufferIssueTracker : IIssueTracker, IIssueConsumer
         private readonly string charset;
         private readonly ILogger logger;
         private readonly IIssuesFilter issuesFilter;
+        private readonly ISonarErrorListDataSource sonarErrorDataSource;
 
         public string FilePath { get; private set; }
-        public SnapshotFactory Factory { get; }
-
-        public IssuesSnapshot LastIssues { get; private set; }
+        internal /* for testing */ SnapshotFactory Factory { get; }
 
         private readonly ISet<IssueTagger> activeTaggers = new HashSet<IssueTagger>();
 
         public TextBufferIssueTracker(DTE dte, TaggerProvider provider, ITextDocument document,
-            IEnumerable<AnalysisLanguage> detectedLanguages, ILogger logger, IIssuesFilter issuesFilter)
-            : this(dte, provider, document, detectedLanguages, new IssueMarkerFactory(), logger, issuesFilter)
+            IEnumerable<AnalysisLanguage> detectedLanguages, IIssuesFilter issuesFilter,
+            ISonarErrorListDataSource sonarErrorDataSource, ILogger logger)
+            : this(dte, provider, document, detectedLanguages, issuesFilter, sonarErrorDataSource, logger, new IssueMarkerFactory())
```

Comment:
Miscellaneous: I've re-ordered the arguments to follow the same convention as other classes:
* in public constructor `ILogger` is last
* in internal constructors, test-only artefacts are added after public arguments.